0023-communication-protocol.adoc

```diff
@@ -0,0 +1,388 @@
+If he doesn't know about a swap with this digest, he closes the substream without sending any message.
+
+You may wonder why Bob generates an ID on top of the digest that Alice shared with him.
+The reason is simple: while highly unlikely, the digest is not unique.
+
+Having Bob generating a random, dedicated identifier for this swap allows us:
+
+.. Make it explicit that Bob agrees to this swap and is knowledgeable of the same parameters as Alice.
+.. Establish a dependency between this libp2p protocol and all others.
+The other libp2p protocols use the swap-id generated by Bob to identify the swap.
+None of the parties can construct a meaningful message without going through this handshake, making it clear that this handshake has to happen before everything else.
+
+=== Sharing the secret hash
+
+Protocol-identifier: `/comit/swap/secrethash/1.0.0`
+
+This protocol is only to be used by Alice, as she is the one generating the secret.
+
+This protocol only includes a single message to share the secret hash.
+If you paid close attention to the protocol identifier, you may have realized that it is actually protocol agnostic:
+It neither mentions HAN, HErc20 or any of the others.
+The reason is simple: We don't need different messages for different protocol combinations if they always just need to exchange a secret hash.
+Using this libp2p protocol implies that the referenced swap is HTLC-based.
+
+==== Secret hash message
+
+[source,json]
+----
+{
+  "swap_id": "713b0c93-53a1-46b2-9418-de8d46750b79",
+  "secret_hash": "122003f2fb5188982860e6bdaa4cebdae5509311da248c4bd7ba1126f78a881cb816"
+}
+----
+
+The `swap_id` refers to the swap-id generated by Bob in the confirmation message.
+The `secret_hash` is a https://github.com/multiformats/multihash[multihash] of the secret generated by Alice.
+
+We use a multi-hash to allow re-use of this message for any HTLC-based protocol, independent of the hash algorithm that is used.
+It is important to note that for a given swap protocol pair, only a single hash algorithm will be valid.
+The above example uses SHA256, which is the hash algorithm we currently use for HAN, HErc20 and HALight.
+
+Implementations MUST verify that the hash algorithm used to encode this multihash is the same one as the one used in the swap protocol.
+Implementations MUST also take care to not being tricked into using a weaker hash algorithm.
+However, this is very unlikely anyway because the hash algorithm used by HTLCs is usually not configurable.
+
+=== Sharing identities
+
+In the context of a swap based on HTLCs, each party has two identities: one on each ledger.
+For the HALight protocol, the parties only need to exchange one identity: their LND pubkey.
+
+We can define very simple protocols to allow for this information to be exchanged.
+These are the protocols we need:
+
+==== Protocols for sharing identities
+
+Protocol identifier: `/comit/swap/identity/bitcoin/1.0.0`
+
+[source,json]
```

Comment:
Nitpick: use Javascript to allow comments.

Reply:
I am using comments, I am using asciidoc syntax. Reader the rendered version :)